solidity/contracts/hooks/OptimismMessageHook.sol

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+import {IOptimismMessageHook} from "../interfaces/hooks/IOptimismMessageHook.sol";
+import {OptimismISM} from "../isms/native/OptimismISM.sol";
+
+// ============ External Imports ============
+import {ICrossDomainMessenger} from "@eth-optimism/contracts/libraries/bridge/ICrossDomainMessenger.sol";
+
+/**
+ * @title OptimismMessageHook
+ * @notice Message hook to inform the Optimism ISM of messages published through
+ * the native Optimism bridge.
+ */
+contract OptimismMessageHook is IOptimismMessageHook {
+    // ============ Constants ============
+
+    // Domain of chain on which the optimism ISM is deployed
+    uint32 public immutable destinationDomain;
```

Comment:
should we have an abstract L2 hook that implements the immutable domain and message destination domain checks?